### PR TITLE
orjson: init at 3.6.3

### PR DESCRIPTION
**Pull request by misuzu**
###### Motivation for this change
`orjson` now compiles using stable rust

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 137969` at 72460530 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.orjson</li>
    <li>python39Packages.orjson</li>
  </ul>
</details>

**Comment by figsoda** [comment]
commit message should probably say `python3Packages.orjson` instead of `orjson`, ofborg is not building anything
@ofborg build python3Packages.orjson

**Comment by misuzu** [comment]
@ofborg build python3Packages.orjson

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64.

Could also add a `pythonImportsCheck` if you like.

**Comment by misuzu** [comment]
@ofborg build python3Packages.orjson

**Comment by figsoda** [comment]
darwin ofborg failed with this error, i'm doing a rebuild
```
warning: ignoring the user-specified setting 'restrict-eval', because it is a restricted setting and you are not a trusted user
warning: unable to download 'https://cache.nixos.org/7f883ccy0k4d891mq52v0jxngj8yb6wk.narinfo': Couldn't resolve host name (6); retrying in 341 ms
warning: unable to download 'https://cache.nixos.org/7f883ccy0k4d891mq52v0jxngj8yb6wk.narinfo': Couldn't resolve host name (6); retrying in 607 ms
warning: unable to download 'https://cache.nixos.org/7f883ccy0k4d891mq52v0jxngj8yb6wk.narinfo': Couldn't resolve host name (6); retrying in 1374 ms
warning: unable to download 'https://cache.nixos.org/7f883ccy0k4d891mq52v0jxngj8yb6wk.narinfo': Couldn't resolve host name (6); retrying in 2556 ms
error: unable to download 'https://cache.nixos.org/7f883ccy0k4d891mq52v0jxngj8yb6wk.narinfo': Couldn't resolve host name (6)
```
@ofborg build python3Packages.orjson

File: pkgs/development/python-modules/orjson/default.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    sha256 = "163rf8k7qb1whjf78i8v5dlyif1sgvd9xkj8li81j45abgzkq86w";
+  };
+
+  format = "pyproject";
+
+  nativeBuildInputs = with rustPlatform; [ cargoSetupHook maturinBuildHook ];
```
Comment (jonringer) [inline pkgs/development/python-modules/orjson/default.nix:36]: Getting a bit long
```suggestion
  nativeBuildInputs = with rustPlatform; [
    cargoSetupHook
    maturinBuildHook
  ];
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 137969` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.orjson</li>
    <li>python39Packages.orjson</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**